### PR TITLE
Add a minimal c++ example in doc

### PR DESCRIPTION
**Pull request by rygx**

### CODE REVIEW

File: docs/cpp.md [hunk 1]
```diff
@@ -3,8 +3,86 @@
...
+```cpp
+#include <iostream>
 
-An example is needed! Please consider contributing one. :pray:
+int main(){
+    std::cout<<"This is a C++ example!"<<std::endl;
+}
+```
+
+To make image and container for this program, the `BUILD.bazel` file would look like this:
```
Comment (thesayyn) [inline docs/cpp.md:17]:
```suggestion
To make a container image for this program, the `BUILD.bazel` one would have something like this:
```

File: docs/cpp.md [hunk 2]
```diff
@@ -3,8 +3,86 @@
...
+
+# Packaging the binary into tar, which is needed by oci_image rule
+pkg_tar(
+    name = "tar",
+    srcs = [":example_binary"],
+)
+
+# Making image
+# C++ programs usually need some fundamental libraries such as glibc, libstdc++, etc.
+# `distroless_base` image does not provide them and C++ programs will complain about
```
Comment (thesayyn) [inline docs/cpp.md:41]: maybe distroless_cc does?

File: docs/cpp.md (hunk 1 above)
Comment (rygx) [reply docs/cpp.md]: Revised as suggested.

File: docs/cpp.md (hunk 2 above)
Comment (rygx) [reply docs/cpp.md]: Makes sense. Changed the example to use `gcr.io/distroless/cc-debian12`

**Comment by rygx** [comment]
@thesayyn is this good to move forward?

**Review by thesayyn — APPROVED** [review]
Thank you!!

**Pull request merged**